### ISSUE TITLE
Running tests in parallel causes non-deterministic results

### ISSUE THREAD

**Issue by sgpthomas** [rachitnigam/runt#8]
I have two test suites using Verilator. Turns out that there is a race condition caused by two threads trying to read the same file at the same time causing one of them to fail. 

```
make: execvp: obj_dir/Vmain: Text file busy
make: *** [Makefile:19: tests/verilog-primitives/register.vcd] Error 127
```

I don't know what to do here. The thing I'll do for now is just make a copy of the contended file so that these can be run in parallel. But maybe there should be an option to run a test suite after another finishes?

**Comment by rachitnigam** [comment]
Can you link to the `runt.toml` file? I don't know if there is a clean way to specify that test suites should run in sequence since the point of runt is to run things in parallel. It assumes that test suites and test don't have any dependencies on each other.

I'm also curious why there is a contended file. Is that a runt issue or a verilator issue?

**Comment by sgpthomas** [comment]
I think this is a Verilator issue and runt shouldn't worry about this. Here's how I fixed it: [https://github.com/cucapra/futil/pull/94/commits/7751b53923fe70271cfdaa93551bb7e5f6893814#diff-b67911656ef5d18c4ae36cb6741b7965](https://github.com/cucapra/futil/pull/94/commits/7751b53923fe70271cfdaa93551bb7e5f6893814#diff-b67911656ef5d18c4ae36cb6741b7965).

Basically, Verilator was generating a folder named `obj_dir` for every test, and so different tests where using the same folder. I fixed it by generating a different `obj_dir` directory for each test. I think this is the right solution.
